go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/cluster-api-provider-vsphere
 
-go 1.21
+go 1.21.8
```

Comment:
```suggestion
go 1.21
```
The go directive shouldn't contain a patch version

Reply:
Ah interesting

Reply:
We'll get back to that when we bump to 1.22 :0

Reply:
Read through this doc: https://go.dev/doc/toolchain. I think e.g. 1.22.0 is fine going forward

> Any two Go versions can be compared to decide whether one is less than, greater than, or equal to the other. If the language versions are different, that decides the comparison: 1.21.9 < 1.22. Within a language version, the ordering from least to greatest is: the language version itself, then release candidates ordered by R, then releases ordered by P.
> 
> For example, 1.21 < 1.21rc1 < 1.21rc2 < 1.21.0 < 1.21.1 < 1.21.2.
> 
> Before Go 1.21, the initial release of a Go toolchain was version 1.N, not 1.N.0, so for N < 21, the ordering is adjusted to place 1.N after the release candidates.

Reply:
I think this explains your issue. 1.22 is simply not an existing toolchain

> The standard Go toolchains are named goV where V is a Go version denoting a beta release, release candidate, or release. For example, go1.21rc1 and go1.21.0 are toolchain names; go1.21 and go1.22 are not (the initial releases are go1.21.0 and go1.22.0), but go1.20 and go1.19 are.

Reply:
Sounds like it, thanks for the clarification 👍